samples/touchtest/touchtest.cpp

```diff
@@ -0,0 +1,246 @@
+
+    int FindIndexOfTouchId(const wxTouchSequenceId& id);
+
+    void DrawStart(const wxTouchSequenceId& id, wxPoint pos);
+    void DrawUpdate(const wxTouchSequenceId& id, wxPoint pos);
+    void DrawEnd(const wxTouchSequenceId& id, wxPoint pos);
+
+public:
+    MyFrame(wxFrame *parent, const wxString& title,
+        const wxPoint& pos, const wxSize& size, const long style);
+
+    void OnTouchBegin(wxMultiTouchEvent& event);
+    void OnTouchMove(wxMultiTouchEvent& event);
+    void OnTouchEnd(wxMultiTouchEvent& event);
+
+    void OnMouseDown(wxMouseEvent& event);
+    void OnMouseMove(wxMouseEvent& event);
+    void OnMouseUp(wxMouseEvent& event);
+
+    void OnPaint(wxPaintEvent& event);
+    void OnSize(wxSizeEvent& event);
+    void OnQuit(wxCommandEvent& event);
+
+DECLARE_EVENT_TABLE()
+};
+
+IMPLEMENT_APP(wxMyApp)
+
+bool wxMyApp::OnInit()
+{
+    if ( !wxApp::OnInit() )
+        return false;
+
+    wxFrame* frame = new MyFrame(nullptr, wxT("Multi-touch Test"), wxDefaultPosition,
+        wxSize(500, 400), wxDEFAULT_FRAME_STYLE | wxHSCROLL | wxVSCROLL);
+    SetTopWindow(frame);
+    frame->CenterOnScreen();
+    frame->Show(true);
+
+    return true;
+}
+
+wxBEGIN_EVENT_TABLE(MyFrame, wxFrame)
+    EVT_MENU(wxID_EXIT, MyFrame::OnQuit)
+    EVT_TOUCH_BEGIN(MyFrame::OnTouchBegin)
+    EVT_TOUCH_MOVE(MyFrame::OnTouchMove)
+    EVT_TOUCH_END(MyFrame::OnTouchEnd)
+    EVT_TOUCH_CANCEL(MyFrame::OnTouchEnd)
+    EVT_LEFT_DOWN(MyFrame::OnMouseDown)
+    EVT_MOTION(MyFrame::OnMouseMove)
+    EVT_LEFT_UP(MyFrame::OnMouseUp)
+
+    EVT_PAINT(MyFrame::OnPaint)
+    EVT_SIZE(MyFrame::OnSize)
+wxEND_EVENT_TABLE()
+
+MyFrame::MyFrame(wxFrame *parent, const wxString& title, const wxPoint& pos,
+    const wxSize& size, const long style)
+    : wxFrame(parent, wxID_ANY, title, pos, size, style),
+    m_MouseId(&m_MouseId)
+{
+    m_TouchPoints[0].pen = wxPen(*wxBLACK, 2);
+    m_TouchPoints[1].pen = wxPen(*wxBLUE, 2);
+    m_TouchPoints[2].pen = wxPen(*wxCYAN, 2);
+    m_TouchPoints[3].pen = wxPen(*wxGREEN, 2);
+    m_TouchPoints[4].pen = wxPen(*wxYELLOW, 2);
+    m_TouchPoints[5].pen = wxPen(*wxRED, 2);
+    m_TouchPoints[6].pen = wxPen(*wxLIGHT_GREY, 2);
+    m_TouchPoints[7].pen = wxPen(*wxBLACK, 2, wxPENSTYLE_DOT_DASH);
+    m_TouchPoints[8].pen = wxPen(*wxBLUE, 2, wxPENSTYLE_DOT_DASH);
+    m_TouchPoints[9].pen = wxPen(*wxCYAN, 2, wxPENSTYLE_DOT_DASH);
+    m_TouchPoints[10].pen = wxPen(*wxGREEN, 2, wxPENSTYLE_DOT_DASH);
+    m_TouchPoints[11].pen = wxPen(*wxRED, 2, wxPENSTYLE_DOT_DASH);
+    m_TouchPoints[12].pen = wxPen(*wxBLACK, 2, wxPENSTYLE_DOT);
+    m_TouchPoints[13].pen = wxPen(*wxYELLOW, 2, wxPENSTYLE_DOT);
+    m_TouchPoints[14].pen = wxPen(*wxBLUE, 2, wxPENSTYLE_DOT);
+    m_TouchPoints[15].pen = wxPen(*wxRED, 2, wxPENSTYLE_DOT);
+
+    wxMenu *file_menu = new wxMenu;
+
+    file_menu->Append(wxID_EXIT);
+
+    wxMenuBar *menu_bar = new wxMenuBar;
+
+    menu_bar->Append(file_menu, wxT("&File"));
+    SetMenuBar(menu_bar);
```

Comment:
This looks rather useless and could be just removed IMO.

Reply:
It turned out to be useful to check if the pos is relative to the client rect, not the window rect.

Reply:
Maybe some options can be added later to the menu.